pkg/client/eventsbuilder.go

```diff
@@ -55,11 +67,11 @@ func WithStartBlock(blockNumber uint64) eventOption {
 }
 
 // WithCheckpointer reads events starting at the position recorded by the checkpointer.
```

Comment:
Might need a tweak to align with the changed function signature.
```suggestion
// WithCheckpointer reads events starting at the checkpoint position.
```